File: crossecore-csharp/Java/Java/Java_/UnresolvedVariableDeclarationFragmentBase.cs
/* CrossEcore is a cross-platform modeling framework that generates C#, TypeScript, 
 * JavaScript, Swift code from Ecore models with embedded OCL (http://www.crossecore.org/).
 * The original Eclipse Modeling Framework is available at https://www.eclipse.org/modeling/emf/.
 * 
 * contributor: Simon Schwichtenberg
 */
 
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using oclstdlib;
using Ecore;
namespace Java_{
	public class UnresolvedVariableDeclarationFragmentBase 
	:VariableDeclarationFragmentImpl, UnresolvedVariableDeclarationFragment
	{
		
		protected override EClass eStaticClass() {
			return Java_PackageImpl.Literals.UNRESOLVEDVARIABLEDECLARATIONFRAGMENT;
		}
		
		
		
		public override Object eGet(int featureID, bool resolve, bool coreType) {
			switch (featureID) {
				case Java_PackageImpl.UNRESOLVEDVARIABLEDECLARATIONFRAGMENT_COMMENTS:
					return comments;
				case Java_PackageImpl.UNRESOLVEDVARIABLEDECLARATIONFRAGMENT_ORIGINALCOMPILATIONUNIT:
					return originalCompilationUnit;
				case Java_PackageImpl.UNRESOLVEDVARIABLEDECLARATIONFRAGMENT_ORIGINALCLASSFILE:
					return originalClassFile;
				case Java_PackageImpl.UNRESOLVEDVARIABLEDECLARATIONFRAGMENT_NAME:
					return name;
				case Java_PackageImpl.UNRESOLVEDVARIABLEDECLARATIONFRAGMENT_PROXY:
					return proxy;
				case Java_PackageImpl.UNRESOLVEDVARIABLEDECLARATIONFRAGMENT_USAGESINIMPORTS:
					return usagesInImports;
				case Java_PackageImpl.UNRESOLVEDVARIABLEDECLARATIONFRAGMENT_EXTRAARRAYDIMENSIONS:
					return extraArrayDimensions;
				case Java_PackageImpl.UNRESOLVEDVARIABLEDECLARATIONFRAGMENT_INITIALIZER:
					return initializer;
				case Java_PackageImpl.UNRESOLVEDVARIABLEDECLARATIONFRAGMENT_USAGEINVARIABLEACCESS:
					return usageInVariableAccess;
				case Java_PackageImpl.UNRESOLVEDVARIABLEDECLARATIONFRAGMENT_VARIABLESCONTAINER:
					return variablesContainer;
			}
			return base.eGet(featureID, resolve, coreType);
		}
		
		
		public override void eSet(int featureID, object newValue) {
			switch (featureID) {
				case Java_PackageImpl.UNRESOLVEDVARIABLEDECLARATIONFRAGMENT_COMMENTS:
					comments.Clear();
					comments.AddRange(((List<EObject>)newValue)?.Cast<Comment>());
					return;
				case Java_PackageImpl.UNRESOLVEDVARIABLEDECLARATIONFRAGMENT_ORIGINALCOMPILATIONUNIT:
					originalCompilationUnit = (CompilationUnit) newValue;
					return;
				case Java_PackageImpl.UNRESOLVEDVARIABLEDECLARATIONFRAGMENT_ORIGINALCLASSFILE:
					originalClassFile = (ClassFile) newValue;
					return;
				case Java_PackageImpl.UNRESOLVEDVARIABLEDECLARATIONFRAGMENT_NAME:
					name = (string) newValue;
					return;
				case Java_PackageImpl.UNRESOLVEDVARIABLEDECLARATIONFRAGMENT_PROXY:
					proxy = (bool) newValue;
					return;
				case Java_PackageImpl.UNRESOLVEDVARIABLEDECLARATIONFRAGMENT_USAGESINIMPORTS:
					usagesInImports.Clear();
					usagesInImports.AddRange(((List<EObject>)newValue)?.Cast<ImportDeclaration>());
					return;
				case Java_PackageImpl.UNRESOLVEDVARIABLEDECLARATIONFRAGMENT_EXTRAARRAYDIMENSIONS:
					extraArrayDimensions = (int) newValue;
					return;
				case Java_PackageImpl.UNRESOLVEDVARIABLEDECLARATIONFRAGMENT_INITIALIZER:
					initializer = (Expression) newValue;
					return;
				case Java_PackageImpl.UNRESOLVEDVARIABLEDECLARATIONFRAGMENT_USAGEINVARIABLEACCESS:
					usageInVariableAccess.Clear();
					usageInVariableAccess.AddRange(((List<EObject>)newValue)?.Cast<SingleVariableAccess>());
					return;
				case Java_PackageImpl.UNRESOLVEDVARIABLEDECLARATIONFRAGMENT_VARIABLESCONTAINER:
					variablesContainer = (AbstractVariablesContainer) newValue;
					return;
			}
			base.eSet(featureID, newValue);
		}
		
		
	}
}

File: crossecore-typescript/java_/UnresolvedVariableDeclarationFragmentBase.ts
import {EClass} from "ecore/EClass";
import {NotificationChain} from "ecore/NotificationChain";
import {ENotificationImpl} from "ecore/ENotificationImpl";
import {NotificationImpl} from "ecore/NotificationImpl";
import {InternalEObject} from "ecore/InternalEObject";
import {BasicEObjectImpl} from "ecore/BasicEObjectImpl";
import {Java_PackageLiterals} from "./Java_PackageLiterals";
import {VariableDeclarationFragmentImpl} from "./VariableDeclarationFragmentImpl";
import {UnresolvedVariableDeclarationFragment} from "./UnresolvedVariableDeclarationFragment";
import {UnresolvedItemImpl} from "./UnresolvedItemImpl";
		
			export class UnresolvedVariableDeclarationFragmentBase
			extends VariableDeclarationFragmentImpl
			implements UnresolvedVariableDeclarationFragment
			{

			
				public static eStaticClass:EClass;
				
				protected eStaticClass():EClass{
					
					return UnresolvedVariableDeclarationFragmentBase.eStaticClass;
				}
			
			
			
				public eGet_number_boolean_boolean(featureID:number, resolve:boolean, coreType:boolean):any{
					switch (featureID) {
						case Java_PackageLiterals.UNRESOLVEDVARIABLEDECLARATIONFRAGMENT_COMMENTS:
							return this.comments;
						case Java_PackageLiterals.UNRESOLVEDVARIABLEDECLARATIONFRAGMENT_ORIGINALCOMPILATIONUNIT:
							return this.originalCompilationUnit;
						case Java_PackageLiterals.UNRESOLVEDVARIABLEDECLARATIONFRAGMENT_ORIGINALCLASSFILE:
							return this.originalClassFile;
						case Java_PackageLiterals.UNRESOLVEDVARIABLEDECLARATIONFRAGMENT_NAME:
							return this.name;
						case Java_PackageLiterals.UNRESOLVEDVARIABLEDECLARATIONFRAGMENT_PROXY:
							return this.proxy;
						case Java_PackageLiterals.UNRESOLVEDVARIABLEDECLARATIONFRAGMENT_USAGESINIMPORTS:
							return this.usagesInImports;
						case Java_PackageLiterals.UNRESOLVEDVARIABLEDECLARATIONFRAGMENT_EXTRAARRAYDIMENSIONS:
							return this.extraArrayDimensions;
						case Java_PackageLiterals.UNRESOLVEDVARIABLEDECLARATIONFRAGMENT_INITIALIZER:
							return this.initializer;
						case Java_PackageLiterals.UNRESOLVEDVARIABLEDECLARATIONFRAGMENT_USAGEINVARIABLEACCESS:
							return this.usageInVariableAccess;
						case Java_PackageLiterals.UNRESOLVEDVARIABLEDECLARATIONFRAGMENT_VARIABLESCONTAINER:
							return this.variablesContainer;
					}
					//return this.eGetFromVariableDeclarationFragment(featureID, resolve, coreType);
					return super.eGet(featureID, resolve, coreType);
				}
				
				public eSet_number_any(featureID:number, newValue:any):void {
					switch (featureID) {
					}
					super.eSet_number_any(featureID, newValue);
				}

				public eBaseStructuralFeatureID(derivedFeatureID:number, baseClass:Function):number {
					if (baseClass === UnresolvedItemImpl) {
						switch (derivedFeatureID) {
							default: return -1;
						}
					}
					return super.eBaseStructuralFeatureID(derivedFeatureID, baseClass);
				}
				
							
				public eDerivedStructuralFeatureID_number_Function(baseFeatureID:number, baseClass:Function):number {
					if (baseClass === UnresolvedItemImpl) {
						switch (baseFeatureID) {
							default: return -1;
						}
					}
					return super.eDerivedStructuralFeatureID_number_Function(baseFeatureID, baseClass);
				}
				
			}
			
